x-pack/plugins/dataset_quality/public/hooks/use_dataset_quality_table.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { orderBy } from 'lodash';
+import React, { useState, useMemo, useCallback } from 'react';
+import { useFetcher } from '@kbn/observability-shared-plugin/public';
+import { tableSummaryAllText, tableSummaryOfText } from '../../common/translations';
+import { DataStreamStat } from '../../common/data_streams_stats/data_stream_stat';
+import { getDatasetQualitTableColumns } from '../components/dataset_quality/columns';
+import { useDatasetQualityContext } from '../components/dataset_quality/context';
+import { useKibanaContextForPlugin } from '../utils';
+
+const DEFAULT_SORT_FIELD = 'title';
+const DEFAULT_SORT_DIRECTION = 'desc';
+type DIRECTION = 'asc' | 'desc';
+type SORT_FIELD = keyof DataStreamStat;
+
+const sortingOverrides: Partial<{ [key in SORT_FIELD]: SORT_FIELD }> = {
+  ['size']: 'sizeBytes',
+};
+
+export const useDatasetQualityTable = () => {
+  const {
+    services: { fieldFormats },
+  } = useKibanaContextForPlugin();
+  const [pageIndex, setPageIndex] = useState(0);
+  const [pageSize, setPageSize] = useState(10);
+  const [sortField, setSortField] = useState<SORT_FIELD>(DEFAULT_SORT_FIELD);
+  const [sortDirection, setSortDirection] = useState<DIRECTION>(DEFAULT_SORT_DIRECTION);
+
+  const { dataStreamsStatsServiceClient: client } = useDatasetQualityContext();
+  const { data = [], loading } = useFetcher(async () => client.getDataStreamsStats(), []);
+
+  const columns = useMemo(() => getDatasetQualitTableColumns({ fieldFormats }), [fieldFormats]);
+
+  const pagination = useMemo(
```

Comment:
Why would we use `useMemo` in this case? afaik we should only use it to avoid rerunning expensive calculations